src/book.py

```diff
@@ -180,39 +217,79 @@ def tokenize(self, on, rem_stopwords=True, stopword_lang='english',
+        non-cont: (Default = False): If True, returns random char/word/sent from starting point
+        randomized: (Default = False): If True, chooses a random point to start making snippets
+        rem_stopwords: Same as in Tokenize
+        ret_as_array: (Default = False): Returns one BookText object by appending all groups of snippets, else returns
+                                         an array of BookText objects
+        random_seed: When positive, generates the same snippets everytime
+        
         """
-        # TODO add ability for random seed
+        if (random_seed is not None):
+            random.seed(random_seed)
+            
         if 'char' in on.lower():
-            assert length < len(self._text)
+            tokens = self._text
         elif 'word' in on.lower():
-            tokens = word_tokenize(self._text)
-            assert length < len(tokens)
+            tokens = self.tokenize('word', rem_stopwords)
         elif 'sent' in on.lower():
-            tokens = sent_tokenize(self._text)
+            tokens = self.tokenize('sent', rem_stopwords)
         else:
             raise KeyError(
-                "Arugument 'on' must refer to character, word, or sentence")
-
-        if non_cont:
-            # TODO add functionality for non-continuous random sampling
-            pass
+                "Argument 'on' must refer to character, word, or sentence")
+            
+        assert groups*length <= len(tokens)
+        return_array = []
+
+        if randomized:
+            start = random.randint(0, len(tokens) - groups*length)
+            tokens = tokens[start:]
+            start = 0
         else:
-            if 'char' in on.lower():
-                start = random.randint(0, len(self._text) - length)
-                snippet = self._text[start:start + length]
-            elif 'word' in on.lower():
-                start = random.randint(0, len(tokens) - length)
-                snippet = (" ").join(tokens[start:start + length])
-            elif 'sent' in on.lower():
-                start = random.randint(0, len(tokens) - length)
-                snippet = (" ").join(tokens[start:start + length])
+            start = 0
+        
+        if non_cont == False:
+            for gr in range(groups):
+                snippet = BookText(rawtext=''.join((" ").join(tokens[length*gr:length*(gr+1)])), 
+                               author = self.author, title = self.title, meta = self.meta)
+                return_array.append(snippet)
+        else:
+            for gr in range(groups): 
+                snippet = BookText(rawtext=''.join((" ").join(tokens[start:start+length])), 
+                               author = self.author, title = self.title, meta = self.meta)
+                snippet._text.rstrip()    
+                return_array.append(snippet)
+                
+                for index in sorted(range(start, start+length), reverse=True):
```

Comment:
Shouldn't this be from start to the length of the text -  (length + start)? That way it spans the whole space of the text rather than just from the starting position to the length of the snippet.

Reply:
No, this is just to drop the char/word/sent that have already appeared in one snippet. There are length number of those only.